### ISSUE TITLE
Monitor data is rewriten if I use twiss calculation during tracking

### ISSUE THREAD

**Issue by YangJiangyan** [xsuite/xsuite#291]
Dear developers,

I am writing for reporting a user feedback about ParticlesMonitor.

I calculated twiss parameters after tracked a few turns,  and then the first row of the monitor data (at_turn=0) was rewritten because of twiss calculation.
  
I inserted a particle monitor into the line at s=0 with the name 'monitor_start':
```
n_part =1
monitor_start = xt.ParticlesMonitor(_context=context,
                                start_at_turn = 0, 
                                stop_at_turn = 1,
                                n_repetitions=10,
                                repetition_period=1,
                                num_particles =n_part)
line.insert_element(index=0, element=monitor_start, name='monitor_start')
tracker = line.build_tracker(_context=context)

```

Initial monitor data:
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>particle_id</th>
      <th>s</th>
      <th>x</th>
      <th>px</th>
      <th>y</th>
      <th>py</th>
      <th>zeta</th>
      <th>delta</th>
      <th>at_element</th>
      <th>at_turn</th>
      <th>state</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>3</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>4</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

Then I started tracking particles using following lines:

```
tracker.track(particles, num_turns=3, turn_by_turn_monitor=True) # track 3 turns
```
output:
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>particle_id</th>
      <th>s</th>
      <th>x</th>
      <th>px</th>
      <th>y</th>
      <th>py</th>
      <th>zeta</th>
      <th>delta</th>
      <th>at_element</th>
      <th>at_turn</th>
      <th>state</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.0</td>
      <td>-0.000839</td>
      <td>-0.000002</td>
      <td>-0.001854</td>
      <td>-0.000014</td>
      <td>0.002173</td>
      <td>0.000035</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>0</td>
      <td>0.0</td>
      <td>-0.000002</td>
      <td>0.000839</td>
      <td>-0.003779</td>
      <td>0.001882</td>
      <td>0.002171</td>
      <td>0.000035</td>
      <td>0</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.000839</td>
      <td>0.000002</td>
      <td>0.001853</td>
      <td>0.000014</td>
      <td>0.002165</td>
      <td>0.000035</td>
      <td>0</td>
      <td>2</td>
      <td>1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>4</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
  </tbody>
</table>
Perform twiss calculation:

```
tracker.twiss(method='4d')  #twiss calculation
```

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>particle_id</th>
      <th>s</th>
      <th>x</th>
      <th>px</th>
      <th>y</th>
      <th>py</th>
      <th>zeta</th>
      <th>delta</th>
      <th>at_element</th>
      <th>at_turn</th>
      <th>state</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.0</td>
      <td>1.000000e-07</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>-0.000100</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>0</td>
      <td>0.0</td>
      <td>-1.657675e-06</td>
      <td>0.000839</td>
      <td>-0.003779</td>
      <td>0.001882</td>
      <td>0.002171</td>
      <td>0.000035</td>
      <td>0</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>0</td>
      <td>0.0</td>
      <td>8.394636e-04</td>
      <td>0.000002</td>
      <td>0.001853</td>
      <td>0.000014</td>
      <td>0.002165</td>
      <td>0.000035</td>
      <td>0</td>
      <td>2</td>
      <td>1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>4</th>
      <td>0</td>
      <td>0.0</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

The data at first row was rewritten, shown in the table above.

I still could get the particle coordinates by using the sampled initial coordinates from xpart.Particles object,
however I think  it would be better if the Monitor data is not overwritten after twiss calculation.

Thank you!

Best regards, 
Jiangyan.

@eltos

**Comment by giadarol** [comment]
It should be easy to handle. Fix coming soon

**Comment by YangJiangyan** [comment]
> 

Many thanks :)

**Comment by giadarol** [comment]
Solved in xsuite v0.26.3
